File: packages/lib-config-py/src/lib_config/http/api/api_base.py
from lib_config.http.api.api_models import ApiConfigModel
from lib_config.http.models import ApiEndpointModel
from lib_shared.http.utils.constants import HTTP_METHOD, HTTP_STATUS_CODE, PING
from lib_shared.http.utils.models import HttpRequestModel, HttpResponseModel


async def ping_handler(request: HttpRequestModel) -> HttpResponseModel:
    return HttpResponseModel(
        status_code=HTTP_STATUS_CODE.OK,
        body="success",
    )


api_config = ApiConfigModel(
    prefix="api",
    routes=[
        ApiEndpointModel(
            pathname=PING,
            method=HTTP_METHOD.GET,
            handler=ping_handler,
        ),
    ],
)

File: packages/lib-config-js/src/api/api.base.ts
import { ping } from '@lib/backend/http/handlers/ping/ping';
import { type ApiConfigModel } from '@lib/config/api/api.models';
import { Config } from '@lib/config/utils/Config/Config';
import { HTTP_METHOD, PING } from '@lib/shared/http/http.constants';

export const apiConfig = new Config<ApiConfigModel>({
  params: () =>
    ({
      prefix: 'api',

      routes: [
        {
          handler: ping,
          method: HTTP_METHOD.GET,
          pathname: PING,
        },
      ],
    }) as ApiConfigModel,
});
